lib/ts/recipe/totp/api/implementation.ts

```diff
@@ -0,0 +1,153 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { APIInterface } from "../";
+import TotpRecipe from "../recipe";
+import SessionError from "../../session/error";
+import MultiFactorAuth from "../../multifactorauth";
+import MultiFactorAuthRecipe from "../../multifactorauth/recipe";
+import { getUser } from "../../..";
+
+export default function getAPIInterface(): APIInterface {
+    return {
+        createDevicePOST: async function ({ deviceName, options, session, userContext }) {
+            const userId = session.getUserId();
+
+            let userIdentifierInfo: string | undefined = undefined;
+            const emailOrPhoneInfo = await TotpRecipe.getInstanceOrThrowError().getUserIdentifierInfoForUserId(
+                session.getUserId(),
+                userContext
+            );
+            if (emailOrPhoneInfo.status === "OK") {
+                userIdentifierInfo = emailOrPhoneInfo.info;
+            } else if (emailOrPhoneInfo.status === "UNKNOWN_USER_ID_ERROR") {
+                throw new SessionError({
+                    type: SessionError.UNAUTHORISED,
+                    message: "Unknown User ID provided",
+                });
+            } else if (emailOrPhoneInfo.status === "USER_IDENTIFIER_INFO_DOES_NOT_EXIST_ERROR") {
+                // Ignore since UserIdentifierInfo is optional
+            }
+
+            let mfaInstance = MultiFactorAuthRecipe.getInstance();
+
+            if (mfaInstance === undefined) {
+                throw new Error("should never come here"); // TOTP can't work without MFA
```

Comment:
```suggestion
                throw new Error("Please initialise the MFA recipe as well."); // TOTP can't work without MFA
```

Reply:
It should never come here because we auto-init MFA. Updated the comment instead of updating the message.